dsn.go

```diff
@@ -66,17 +68,48 @@ type Config struct {
 	MultiStatements          bool // Allow multiple statements in one query
 	ParseTime                bool // Parse time values to time.Time
 	RejectReadOnly           bool // Reject read-only connections
+
+	// private fields. new options should be come here
+
+	pubKey       *rsa.PublicKey // Server public key
+	timeTruncate time.Duration  // Truncate time.Time values to the specified duration
 }
 
+// Functional Options Pattern
+// https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+type option func(*Config) error
+
 // NewConfig creates a new Config and sets default values.
-func NewConfig() *Config {
-	return &Config{
+func NewConfig(opts ...option) *Config {
+	cfg := &Config{
```

Comment:
My note:

Changing the signature may cause a compile error.
For example:

```
var f func() mysql.Config
f = mysql.NewConfig
```

I searched on GitHub, but I could not find any code that uses it in the way described in the example.
https://github.com/search?q=mysql.NewConfig+language%3AGo&type=code&l=Go

Therefore, the impact of changing the signature is considered to be minor.